cmd/mattermost/commands/plugin.go

```diff
@@ -0,0 +1,181 @@
+// Copyright (c) 2018-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package commands
+
+import (
+	"errors"
+	"os"
+
+	"github.com/spf13/cobra"
+)
+
+var PluginCmd = &cobra.Command{
+	Use:   "plugin",
+	Short: "Management of plugins",
+}
+
+var PluginCreateCmd = &cobra.Command{
+	Use:     "add",
+	Short:   "Add a plugin",
+	Long:    "Add a plugin or multiple plugins",
+	Example: `  plugin add hovercardexample.tar.gz pluginexample.tar.gz`,
+	RunE:    pluginCreateCmdF,
+}
+
+var PluginDeleteCmd = &cobra.Command{
+	Use:     "delete",
+	Short:   "Delete a plugin",
+	Long:    "Delete a plugin",
+	Example: `  plugin delete hovercardexample`,
+	RunE:    pluginDeleteCmdF,
+}
+
+var PluginEnableCmd = &cobra.Command{
+	Use:     "enable",
+	Short:   "Enable a plugin",
+	Long:    "Enable a plugin or multiple plugins",
+	Example: `  plugin enable hovercardexample.tar.gz pluginexample.tar.gz`,
+	RunE:    pluginEnableCmdF,
+}
+
+var PluginDisableCmd = &cobra.Command{
+	Use:     "disable",
+	Short:   "Disable a plugin",
+	Long:    "Disable a plugin",
+	Example: `  plugin disable hovercardexample`,
+	RunE:    pluginDisableCmdF,
+}
+
+var PluginListCmd = &cobra.Command{
+	Use:     "list",
+	Short:   "List plugins",
+	Long:    "List all active and inactive plugins",
+	Example: `  plugin list`,
+	RunE:    pluginListCmdF,
+}
+
+func init() {
+	PluginCmd.AddCommand(
+		PluginCreateCmd,
+		PluginDeleteCmd,
+		PluginEnableCmd,
+		PluginDisableCmd,
+		PluginListCmd,
+	)
+	RootCmd.AddCommand(PluginCmd)
+}
+
+func pluginCreateCmdF(command *cobra.Command, args []string) error {
+	a, err := InitDBCommandContextCobra(command)
+	if err != nil {
+		return err
+	}
+	defer a.Shutdown()
+
+	if len(args) < 1 {
+		return errors.New("Expected at least one argument. See help text for details.")
+	}
+
+	for i, plugin := range args {
+		fileReader, err := os.Open(plugin)
+		if err != nil {
+			return err
+		}
+
+		if _, err := a.InstallPlugin(fileReader); err != nil {
+			return errors.New("Unable to create plugin: " + args[i] + ". Error: " + err.Error())
+		}
+		fileReader.Close()
```

Comment:
It is not a big deal, because the app will exit and close it automatically, but if the InstallPlugin fails the current plugin file is not closed.

Reply:
I can `defer fileReader.Close()` after opening the file.